packages/amplify-category-auth/resources/auth-custom-resource/hostedUIProviderLambda.js

```diff
@@ -12,17 +13,53 @@
   try {
     const userPoolId = event.ResourceProperties.userPoolId;
     const hostedUIProviderMeta = JSON.parse(event.ResourceProperties.hostedUIProviderMeta);
+    const hostedUIProviderCreds = JSON.parse(event.ResourceProperties.hostedUIProviderCreds);
+    const stackName = event.ResourceProperties.stackName;
 
-    for (const { ProviderName } of hostedUIProviderMeta) {
-      try {
-        const params = { ProviderName, UserPoolId: userPoolId };
-        await identity.deleteIdentityProvider(params).promise();
-      } catch (e) {
-        if (!e?.code?.toString()?.match(/NotFoundException/)) {
-          // bubble up to outer catch.
+    for (const providerMeta of hostedUIProviderMeta) {
+      const providerCreds = hostedUIProviderCreds.find(({ ProviderName }) => providerMeta.ProviderName === ProviderName);
+      if (providerMeta.ProviderName === 'SignInWithApple' && !providerCreds.private_key && event.RequestType === 'Update') {
+        try {
+          const params = {
+            UserPoolId: userPoolId,
+            ProviderName: providerMeta.ProviderName,
+            AttributeMapping: providerMeta.AttributeMapping,
+          };
+
+          if (providerCreds.client_id && providerCreds.team_id && providerCreds.key_id && providerCreds.private_key) {
+            params.ProviderDetails = {
+              client_id: providerCreds.client_id,
+              team_id: providerCreds.team_id,
+              key_id: providerCreds.key_id,
+              private_key: providerCreds.private_key,
+              authorize_scopes: providerMeta.authorize_scopes,
+            };
+          }
+
+          await identity.updateIdentityProvider(params).promise();
+        } catch (e) {
           throw e;
-        } else {
-          console.log('Not Found', ProviderName);
+        }
+      } else {
+        try {
+          if (stackName) {
+            const { StackResources } = await cloudformation.describeStackResources({ StackName: stackName }).promise();
+            const resource = StackResources.find((resource) => {
+              resource.LogicalResourceId === `HostedUI${providerMeta.ProviderName}ProviderResource`;
```

Comment:
I'm confused as to why this warning is present in this PR because I don't see the file touched, but the warning looks legitimate.

Reply:
I think prettier changed that, but now that I am reading this, without an explicit `return`, this will return undefined. I'll need to change this and figure out why prettier keeps creating this statement for me.

Reply:
I updated it in the latest commit